src/managers/quorum.py

```diff
@@ -228,8 +228,21 @@ def update_acls(self, event: RelationEvent | None = None) -> None:
             logger.debug(f"UPDATE CHROOT - {client.database}")
             self.client.set_acls_znode_leader(path=client.database, acls=[generated_acl])
 
+            subnodes = self.client.leader_znodes(path=client.database)
+            for node in subnodes:
+                self.client.set_acls_znode_leader(path=node, acls=[generated_acl])
+
         # Looks for applications no longer in the relation but still in config
+        restricted_acl = make_acl(
```

Comment:
**question:** Am I correct in assuming that this is supposed to 'overwrite' existing ACLs, making it `super` only?

If so, have you checked that it does that? I only ask as you can have multiple ACLs for a given zNode. I know that Kazoo interprets this as a list, but I don't know if that list is applied as a 'diff' (aka only updating the `super` ACL, but leaving all the others), or as an 'overwrite'.

Reply:
Yes, it does overwrite the existing ACLs. To extend the existing ones, we would need to first get them, and then pass the complete list we want